pyccel/codegen/printing/cwrappercode.py

```diff
@@ -750,10 +757,11 @@ def _print_FunctionDef(self, expr):
         used_names = set([a.name for a in expr.arguments] + [r.name for r in expr.results] + [expr.name])
 
         # update ndarray local variables properties
-        local_arg_vars = [a.clone(a.name, is_pointer=True, allocatable=False)
-                          if isinstance(a, Variable) and a.rank > 0 else a for a in expr.arguments]
+        local_arg_vars = {(a.var.clone(a.var.name, is_pointer=True, allocatable=False)
+                          if isinstance(a.var, Variable) and a.var.rank > 0 else a.var):a for a in expr.arguments}
         # update optional variable properties
-        local_arg_vars = [a.clone(a.name, is_pointer=True) if a.is_optional else a for a in local_arg_vars]
+        local_arg_vars = {(v.clone(v.name, is_pointer=True) if v.is_optional \
+                            else v) : a for v,a in local_arg_vars.items()}
```

Comment:
The property `a.var` is used a lot here, so it is probably beneficial to extract the arguments first. Further the two dictionary comprehensions can probably be merged, given that every ndarray local variable is already a pointer, although that looks a bit awkward. Could something like this work?
```python
arg_vars = {a.var : a for a in expr.arguments}
local_arg_vars = {(v.clone(v.name, is_pointer=True, allocatable=False)
    if isinstance(v, Variable) and v.rank > 0 else (v.clone(v.name, is_pointer=True)
    if v.is_optional else v)) : a for v, a in arg_vars.items()}
```

Reply:
Fixed d254d8b3

Reply:
> Fixed [d254d8b](https://github.com/pyccel/pyccel/commit/d254d8b35806a714864edd293d0a18d0b07494e6)

Nice, that's even shorter!